src/editors/autocompleteEditor.js

```diff
@@ -77,247 +41,329 @@ function onBeforeKeyDown(event) {
-  dropdownStyle.top = `${-dropdownHeight}px`;
+  highlightBestMatchingChoice(index) {
+    if (typeof index === 'number') {
+      this.htEditor.selectCell(index, 0, void 0, void 0, void 0, false);
+    } else {
+      this.htEditor.deselectCell();
+    }
+  }
 
-  this.htEditor.flipped = true;
-};
+  getDropdownHeight() {
+    const firstRowHeight = this.htEditor.getInstance().getRowHeight(0) || 23;
+    const visibleRows = this.cellProperties.visibleRows;
 
-AutocompleteEditor.prototype.unflipDropdown = function() {
-  const dropdownStyle = this.htEditor.rootElement.style;
+    return this.strippedChoices.length >= visibleRows ? (visibleRows * firstRowHeight) : (this.strippedChoices.length * firstRowHeight) + 8;
+  }
 
-  if (dropdownStyle.position === 'absolute') {
-    dropdownStyle.position = '';
-    dropdownStyle.top = '';
+  stripValueIfNeeded(value) {
+    return this.stripValuesIfNeeded([value])[0];
   }
 
-  this.htEditor.flipped = void 0;
-};
+  stripValuesIfNeeded(values) {
+    const { allowHtml } = this.cellProperties;
 
-AutocompleteEditor.prototype.updateDropdownHeight = function() {
-  const currentDropdownWidth = this.htEditor.getColWidth(0) + getScrollbarWidth() + 2;
-  const trimDropdown = this.cellProperties.trimDropdown;
+    const stringifiedValues = arrayMap(values, value => stringify(value));
+    const strippedValues = arrayMap(stringifiedValues, value => (allowHtml ? value : stripTags(value)));
 
-  this.htEditor.updateSettings({
-    height: this.getDropdownHeight(),
-    width: trimDropdown ? void 0 : currentDropdownWidth
-  });
+    return strippedValues;
+  }
 
-  this.htEditor.view.wt.wtTable.alignOverlaysWithTrimmingContainer();
-};
+  allowKeyEventPropagation(keyCode) {
+    const selectedRange = this.htEditor.getSelectedRangeLast();
+    const selected = { row: selectedRange ? selectedRange.from.row : -1 };
+    let allowed = false;
 
-AutocompleteEditor.prototype.setDropdownHeight = function(height) {
-  this.htEditor.updateSettings({
-    height
-  });
-};
+    if (keyCode === KEY_CODES.ARROW_DOWN && selected.row > 0 && selected.row < this.htEditor.countRows() - 1) {
+      allowed = true;
+    }
+    if (keyCode === KEY_CODES.ARROW_UP && selected.row > -1) {
+      allowed = true;
+    }
 
-AutocompleteEditor.prototype.finishEditing = function(restoreOriginalValue, ...args) {
-  HandsontableEditor.prototype.finishEditing.apply(this, [restoreOriginalValue, ...args]);
-};
+    return allowed;
+  }
 
-AutocompleteEditor.prototype.highlightBestMatchingChoice = function(index) {
-  if (typeof index === 'number') {
-    this.htEditor.selectCell(index, 0, void 0, void 0, void 0, false);
-  } else {
-    this.htEditor.deselectCell();
+  close(...args) {
+    this.instance.removeHook('beforeKeyDown', onBeforeKeyDown);
+    super.close(...args);
   }
-};
+
+  discardEditor(...args) {
+    super.discardEditor(...args);
+
+    this.instance.view.render();
+  }
+}
 
 /**
  * Filters and sorts by relevance
```

Comment:
I think the `sortByRelevance` function could be declared as a `static` inside the class. (It should eventually be moved to a helper file, but it would be a breaking change, so let's skip it for now)